sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py

```diff
@@ -535,3 +537,77 @@ def _to_generated(self) -> GeneratedConfigurationSnapshot:
             retention_period=self.retention_period,
             tags=self.tags,
         )
+
+
+class ConfigurationSettingPropertiesPaged(PageIterator):
+    """An iterable of ConfigurationSetting properties."""
+
+    page_etag: str
+    """The etag of current page."""
+    
+    def __init__(self, command: Callable, **kwargs: Any) -> None:
+        super(ConfigurationSettingPropertiesPaged, self).__init__(
+            self._get_next_cb,
+            self._extract_data_cb,
+            continuation_token=kwargs.get("continuation_token"),
+        )
+        self._command = command
+        self._key = kwargs.get("key")
+        self._label = kwargs.get("label")
+        self._accept_datetime = kwargs.get("accept_datetime")
+        self._select = kwargs.get("select")
+        self._cls = kwargs.get("cls")
+
+    
+    def _get_next_cb(self, continuation_token):  # pylint: disable=inconsistent-return-statements
+        response = self._command(
+            key=self._key,
+            label=self._label,
+            accept_datetime=self._accept_datetime,
+            select=self._select,
+            cls=self._cls,
+        ).by_page(continuation_token=continuation_token)
+        next(response)
+        self.page_etag = response._response.http_response.headers['Etag']
+        return response
+
+
+    def _extract_data_cb(self, pipeline_response):
+        return pipeline_response.continuation_token, pipeline_response._current_page
+
+
+class ConfigurationSettingPropertiesPagedAsync(AsyncPageIterator):
+    """An iterable of ConfigurationSetting properties."""
+
+    page_etag: str
+    """The etag of current page."""
+    
+    def __init__(self, command: Callable, **kwargs: Any) -> None:
+        super(ConfigurationSettingPropertiesPagedAsync, self).__init__(
+            self._get_next_cb,
+            self._extract_data_cb,
+            continuation_token=kwargs.get("continuation_token"),
+        )
+        self._command = command
+        self._key = kwargs.get("key")
+        self._label = kwargs.get("label")
+        self._accept_datetime = kwargs.get("accept_datetime")
+        self._select = kwargs.get("select")
+        self._cls = kwargs.get("cls")
+
+    
+    async def _get_next_cb(self, continuation_token):  # pylint: disable=inconsistent-return-statements
+        response = self._command(
+            key=self._key,
+            label=self._label,
+            accept_datetime=self._accept_datetime,
+            select=self._select,
+            cls=self._cls,
+        ).by_page(continuation_token=continuation_token)
+        await anext(response)
+        self.page_etag = response._response.http_response.headers['Etag']
```

Comment:
Do we really need to reach into to innards to get access to headers?

Reply:
Also, do we want to treat the `ETag` as required? I believe it would be better to not raise here if the ETag is missing...

Reply:
Changed to get the value by `headers.get("Etag")`.